include/oneapi/dpl/pstl/hetero/dpcpp/unseq_backend_sycl.h

```diff
@@ -235,6 +235,32 @@ struct reduce
     {
         return reduce_impl(__item_id, __global_idx, __n, __local_mem, __has_known_identity<_BinaryOperation1, _Tp>{});
     }
+
+    template <typename _Result>
+    auto
+    apply_init(_Result __result) const
+    {
+        return __result; //no op
+    }
+};
+
+// Reduce on local memory
+template <typename _ExecutionPolicy, typename _BinaryOperation1, typename _Tp>
+struct reduce_init : public reduce<_ExecutionPolicy, _BinaryOperation1, _Tp>
```

Comment:
Instead of defining another reduction brick and adding a public method to a functor type, cannot the difference of "init vs. no init" handled in a way similar to `__scan_init_processing` below in this file, or maybe even unified (with some renaming)? I like the scan approach more, and I certainly want to use a single approach, not two distinct ones.

Reply:
Yes, we can use the stuff for scan as unified with renaming:
`__scan_init_processing`  => `__init_processing` 
`__scan_no_init` => `__no_init_value`
`__scan_init `=> `__init_value`